routing/router.go

```diff
@@ -908,6 +905,14 @@ func (r *ChannelRouter) networkHandler() {
 	graphPruneTicker := time.NewTicker(r.cfg.GraphPruneInterval)
 	defer graphPruneTicker.Stop()
 
+	// We'll set up a single tick for first time pruning of the channel
+	// graph in case we are running with assumechanvalid. This is to avoid
+	// doing this at startup, slowing down startup of lnd.
+	var firstTimePrune <-chan time.Time
```

Comment:
Wondering whether we can fold this in with `graphPruneTicker` so that we don't need a special case in the select which has the same logic. Perhaps using `ticker.Reset` but that might be more convoluted than just ticking once with a special case.